include/wimlib/compiler.h

```diff
@@ -127,10 +139,13 @@
 #define max3(a, b, c)	max(max((a), (b)), (c))
 
 /* Swap the values of two variables, without multiple evaluation.  */
+#ifdef _MSC_VER
+#  define swap(a, b, type) { type _a = (a); (a) = (b); (b) = _a; }
+#endif
 #ifndef swap
-#  define swap(a, b) ({ typeof(a) _a = (a); (a) = (b); (b) = _a; })
```

Comment:
Latest Visual Studio supports `typeof()` when compiling with the `/std:clatest` flag, so you can avoid the whole dance around `typeof()` and keep the `swap(a, b)` as it is. This also applies to other uses of `typeof()` in the codebase.